ansible_collections/arista/avd/roles/eos_validate_state/tasks/main.yml

```diff
@@ -50,6 +50,39 @@
   tags:
     - always
 
+- name: Include existing ANTA catalog
+  ansible.builtin.include_vars:
+    file: "{{ input_catalog_path }}"
+    name: input_catalog
+  delegate_to: localhost
+  when: input_catalog_path is defined
+  run_once: true
+  tags:
+    - anta
+
+- name: Generate ANTA catalog
+  arista.avd.anta_catalog_builder:
+    input_catalog: "{{ input_catalog | default(None) }}"
```

Comment:
Moving large data blobs into Ansible will only slow things down. I think we should just pass the path to the action, and then we can load the input_catalog in there. Alternatively use a lookup plugin to load inline in this parameter.

Also collapsing the `anta_catalog_builder` and `anta_runnner` would save us from getting all the catalog vars set in Ansible. You could save the catalog still in a file for testing/dryrun

Reply:
```suggestion
    input_catalog: "{{ (lookup('ansible.builtin.file', input_catalog_path) | from_yaml) if input_catalog_path is arista.avd.defined }}"
```